sympy/tensor/array/dense_ndim_array.py

```diff
@@ -157,6 +157,10 @@ def _eval_simplify(self, **kwargs):
         from sympy.simplify.simplify import simplify
         return self.applyfunc(simplify)
 
+    def evalf(self, **kwargs):
+        f=lambda x: x.evalf()
+        return self.applyfunc(f)
```

Comment:
Rather than defining `evalf` then `EvalfMixin` class should be subclassed somewhere.

Reply:
> Rather than defining `evalf` then `EvalfMixin` class should be subclassed somewhere.

Hi, thanks for the feedback. As I'm not familiar with this code base I'm doing a bit of digging based on your suggestion, but if you have an example in mind for me to review, I can attempt to match that. I did find this example in the matrices section that is similar to my original implementation, but supports more options than my patch:

https://github.com/sympy/sympy/blob/master/sympy/matrices/common.py#L2027

Reply:
There are several examples:
```console
$ git grep EvalfMixin
doc/src/contributing/dependencies.md:  [evalf](sympy.core.evalf.EvalfMixin.evalf).
doc/src/explanation/glossary.md:    [*`evalf`*](sympy.core.evalf.EvalfMixin.evalf) is the method present on
doc/src/explanation/glossary.md:    <sympy.core.evalf.EvalfMixin.n>` method and {func}`~.N` function are both
doc/src/guides/assumptions.rst:number with :py:meth:`~.EvalfMixin.evalf`:
doc/src/guides/assumptions.rst:call to :py:meth:`~.EvalfMixin.evalf` will fully evaluate. The
doc/src/modules/core.rst:.. autoclass:: EvalfMixin
sympy/core/basic.py:        sympy.core.evalf.EvalfMixin.evalf: calculates the given formula to a desired level of precision
sympy/core/evalf.py:        ``EvalfMixin.evalf`` and in addition,
sympy/core/evalf.py:class EvalfMixin:
sympy/core/expr.py:from .evalf import EvalfMixin, pure_complex, DEFAULT_MAXPREC
sympy/core/expr.py:class Expr(Basic, EvalfMixin):
sympy/core/relational.py:from .evalf import EvalfMixin
sympy/core/relational.py:class Relational(Boolean, EvalfMixin):
sympy/geometry/entity.py:from sympy.core.evalf import EvalfMixin, N
sympy/geometry/entity.py:class GeometryEntity(Basic, EvalfMixin):
sympy/geometry/tests/test_entity.py:def test_geometry_EvalfMixin():
sympy/physics/control/lti.py:from sympy.core.evalf import EvalfMixin
sympy/physics/control/lti.py:class LinearTimeInvariant(Basic, EvalfMixin):
sympy/physics/vector/dyadic.py:from sympy.core.evalf import EvalfMixin
sympy/physics/vector/dyadic.py:class Dyadic(Printable, EvalfMixin):
sympy/physics/vector/vector.py:from sympy.core.evalf import EvalfMixin
sympy/physics/vector/vector.py:class Vector(Printable, EvalfMixin):
sympy/sets/sets.py:from sympy.core.evalf import EvalfMixin
sympy/sets/sets.py:class Set(Basic, EvalfMixin):
```